base/printf.jl

```diff
@@ -38,24 +38,23 @@ end
 function parse(s::AbstractString)
     # parse format string into strings and format tuples
     list = []
-    i = j = start(s)
-    j1 = 0 # invariant: j1 == prevind(s, j)
-    while !done(s,j)
-        c, k = next(s,j)
+    a = Iterators.Stateful(pairs(s))
+    lastparse = firstindex(s)
+    lastidx = 0 # invariant: lastidx == prevind(s, idx)
+    for (idx, c) in a
         if c == '%'
-            i > j1 || push!(list, s[i:j1])
-            flags, width, precision, conversion, k = parse1(s,k)
+            lastparse > lastidx || push!(list, s[lastparse:lastidx])
```

Comment:
SubString?

Reply:
I mean, sure, but that's unrelated to this PR.